### ISSUE TITLE
bug: NoKoinComponentInterface skips non-companion object declarations

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#25]
## Description

`NoKoinComponentInterface.visitObjectDeclaration()` has `if (declaration.isCompanion())` guard, so regular `object` declarations implementing `KoinComponent` are not detected.

## Reproduction

```kotlin
object MySingleton : KoinComponent {  // NOT DETECTED
    val service = get<MyService>()
}
```

**Expected:** Rule reports KoinComponent usage  
**Actual:** No finding

## Fix

Remove the `isCompanion()` guard — check all object declarations.
